src/regfox/regfox_api.js

```diff
@@ -79,15 +81,45 @@ const login = async (email, password) => {
   }).then(response => response.json())
     .then(response => {
       if (get(response, 'data.mutationResponse.errors')) {
-        throw new Error(response);
+        throw new Error(`Could not parse or process response, see details to fix here: ${JSON.stringify(response)}`);
       }
 
       if (!get(response, 'data.mutationResponse.success', false)) {
-        throw new Error(response);
+        throw new Error(`Could not parse or process response, see details to fix here: ${JSON.stringify(response)}`);
       }
 
       return response.data.mutationResponse;
     });
 }
 
-export { searchRegistrations, exchangeBearerToken, login, REGFOX_GRAPHQL_URL, REGFOX_EXCHANGE_TOKEN_URL };
\ No newline at end of file
+/**
+ * Returns a nicely parsed object containing registration info given an id.
+ * Returns a promise Error if the network is down or parsing failed.
+ * 
+ * @param {*} id of a registration, note this is **not** the RegistrationId, its just the id
+ */
+const getRegistrationInfo = async (id) => {
+  const fullUrl = getRegistrationInfoUrl(id);
+  return fetch(fullUrl, {
+    method: 'GET',
+    headers: buildHeaders(undefined)
+  }).then(response => response.json())
+    .then(response => {
```

Comment:
In C# the equivalent of `.then()` is the `.ContinueWith()` method. It's become an antipattern over time as everyone suggests using async/await operations instead of a bunch of chaining. There's some specific reasons why in C# related to how the threading model will end up handling those that are related, but it's primarily related to keeping the code neat.

Any thoughts on that here? Awaiting individual operations vs chaining them together?

Reply:
Definitely never mix the two with something like `.then(async () => await...)`. Otherwise, I have no strong feelings one way or another.

In a case like this where we would have a bunch of extra variables laying about (`response`, `jsonResponse`, `jsonDataResponse`, etc) as we process things down to the final "response" I think `.then` is slightly nicer in that those intermediates don't get exposed unnecessarily.